File: Maths_And_Stats/Perfect_Number/perfectnumber.php
<?php 

function isPerfectNumber($N) 
{
	$sum = 0; 
	for ($i = 1; $i < $N; $i++) 
	{ 
		if ($N % $i == 0) 
		{ 
			$sum = $sum + $i; 
		}	 
	} 
	

	return $sum == $N; 
} 

// Driver's code 
$N = 6; 

if (isPerfectNumber($N)) 
	echo " Perfect Number"; 
else
	echo "Not Perfect Number"; 
?> 

File: Maths_And_Stats/Perfect_Number/perfectnumber.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main()
{
  int i, n, num, sum = 0;
  cout << "Enter The Number" << endl;
  cin >> num;

  for (i = 1; i <= sqrt(num); i++)
  {
    if (num % i == 0)
    {
      if (num / i == 1)
      {
        sum = 0;
      }
      else
      {
        sum = sum + i;
      }
      if (num / i != i && num / i != num)
      {
        sum = sum + (num / i);
      }
    }
  }
  if (sum == num)
  {
    printf("PERFECT NUMBER  \n");
  }
  else
  {
    printf("NOT A PERFECT NUMBER  \n");
  }
  return 0;
}
